oeps/oep-0011.rst

```diff
@@ -0,0 +1,412 @@
+   the Developer's Guide chapter `edX Accessibility Guidelines`_.
+
+9. **Front end applications should support localization**
+
+   Rationale: Open edX is used by learners all over the world, and so
+   it is critical that all new functionality support localization. This means
+   that every string in an application should be marked for translation,
+   and it also includes ensuring that all new user interfaces will switch to
+   right-to-left mode when the language requires it.
+
+   You can learn more about localization with the Developer's Guide
+   chapter `edX Internationalization`_.
+
+10. **Front end implementations should support theming**
+
+   Rationale: Open edX is used by organizations around the world,
+   and each one requires that the user experience is based upon their
+   own brand. This includes elements as simple as overriding the fonts,
+   imagery and color schemes, to more complex customization of headers,
+   footers and registration flows.
+
+   You can learn more about how Open edX supports theming with
+   the guide `Changing Themes for an Open edX Site`_. One simple way to
+   make your interface themeable is to use patterns from the
+   `edX Pattern Library`_ as they already support theming.
+
+11. **Front end code should be safe from cross site scripting**
+
+   Cross Site Scripting (XSS) vulnerabilities allow user-supplied data to be
+   incorrectly executed as code in a web browser. This can allow an attacker
+   to inject malicious scripts, which in the worst case can cause the loss
+   of private user data.
+
+   In order to keep your code safe from such attackers, you should follow the
+   best practices laid out in the guide
+   `Preventing Cross Site Scripting Vulnerabilities`_.
+
+Technology Selection
+~~~~~~~~~~~~~~~~~~~~
+
+1. **Server-side content should be rendered with Django Templates**
+
+   Rationale: There are many template languages available for Django,
+   but the simplest option is to use the built-in Django template engine.
+   The Open edX codebase has a mixture of Django and Mako templates, but the
+   former are easier to reason about because they don't support arbitrary
+   code evaluation. See `Reasons for rejecting Mako`_ for a more detailed
+   explanation.
+
+   **Exception**: Mako templates can continue to be used within edx-platform
+   for consistency with the existing code base. This is because the base
+   templates and theming support are all provided via Mako, so it is too
+   much to expect a new feature to be implemented with a different framework.
+   There is much desire to replace Mako within edx-platform so this
+   exception may eventually be removed.
+
+2. **ECMAScript 2015 should be the supported version of JavaScript**
+
+   Rationale: ECMAScript is the standardized version of JavaScript,
+   and modern front end libraries typically use ES2015 (aka ES6).
+   Most browsers now provide native support for ES2015, but to support
```

Comment:
We have a practice for amending OEPs. We could call this out as something that will be shifting, and do amendments as we decide to upgrade. I'd rather keep it updated through explicit changes though, and not say that we'll follow the spec in lock step (even if that's our general intent).